src/api/index.js

```diff
@@ -1,56 +1,49 @@
-import RollAuthentication from "./authentication";
-import RollUserAPI from "./user";
-import RollRequest from "./request";
-import RollTransactionAPI from "./transaction";
+import { auth } from "@tryrolljs/sdk";
+import { client } from "@tryrolljs/api";
 import config from "../config";
 
-class RollAPI {
-  constructor({ apiURL, oauthConfig, authCacheConfig }) {
-    this.authentication = new RollAuthentication(oauthConfig, authCacheConfig);
-    const request = new RollRequest(this.authentication.getAuthToken, apiURL);
-    this.user = new RollUserAPI(this.authentication.getAuthToken, request);
-    this.transaction = new RollTransactionAPI(
-      this.authentication.getAuthToken,
-      request
-    );
-  }
-}
-
-const OAUTH_KEY = "oauthTokens";
-
-// Implement custom functions to cache tokens between sessions
-const cacheOauthTokens = (oauthTokens) => {
-  localStorage.setItem(OAUTH_KEY, oauthTokens);
+const authSdkConfig = {
+  clientId: config.clientID,
+  issuerUrl: config.issuerURL,
+  redirectUrl: config.redirectURL,
+  logoutRedirectUrl: config.redirectURL,
+  scopes: config.scopes,
 };
 
-const getCachedOauthTokens = () => {
-  return localStorage.getItem(OAUTH_KEY);
-};
+const authSdk = new auth.AuthSDK(authSdkConfig, localStorage);
 
-const clearCachedOauthTokens = () => {
-  localStorage.removeItem(OAUTH_KEY);
+const apiClientConfig = {
+  getApiUrl: () => config.apiURL,
+  getAuthorization: () => {
```

Comment:
the consumer shouldn't be responsible for defining how the auth token is formatted

Reply:
can we set this up such that the api package expects an authorization interface, which the auth sdk fulfills out of the box? Similar to the way that the auth sdk expects a persistence interface, which local storage fulfills.